### PR TITLE
Enable require_trailing_commas lint rule

### PR DESCRIPTION
**Pull request by chenzhiguang**
Add this optional trailing commas to parameter lists to get good formatting. It is also recommended in [Flutter: Using trailing commas
](https://docs.flutter.dev/development/tools/formatting#using-trailing-commas)

### CODE REVIEW

**Comment by kevmoo** [comment]
I like this lint, but since @srawlins is the primary maintainer here, I'll let him comment!

File: benchmark/benchmark.dart
Comment (srawlins) [inline benchmark/benchmark.dart:61]: I am not a fan of this lint rule. For cases like this.

I am a fan of adding trailing commas to argument lists when there is more than one argument, and they span more than 1 line without the comma.

This change here is harder for me to read though.

File: bin/markdown.dart
Comment (srawlins) [inline bin/markdown.dart:22]: I like this change though.

File: bin/markdown.dart
Comment (srawlins) [inline bin/markdown.dart:82]: Again, I don't think this is helpful.

**Review by srawlins — COMMENTED** [review]
I like many of the changes that this induces, but I strongly dislike some of them. What do you say to keeping many of these changes (I can mark the ones to revert), and we do not change analysis_options.yaml?

**Comment by chenzhiguang** [comment]
> I like many of the changes that this induces, but I strongly dislike some of them. What do you say to keeping many of these changes (I can mark the ones to revert), and we do not change analysis_options.yaml?

I do not like some of them too. 

Yep. let me remove the rule from analysis_options.yaml, and reformat it.

File: lib/src/block_parser.dart
Comment (srawlins) [inline lib/src/block_parser.dart:56]: I think this should be reverted.

File: lib/src/inline_parser.dart
Comment (srawlins) [inline lib/src/inline_parser.dart:1155]: I think this should be reverted. It just feels very strange to me in cases like this where there is only one parameter...

File: tool/stats_lib.dart
Comment (srawlins) [inline tool/stats_lib.dart:18]: I think this should be reverted.

But this change I think indicates a smell; this arrow-function should probably be a function body with a local variable, like a variable for the `reflect` result...

File: tool/update_emojis.dart
Comment (srawlins) [inline tool/update_emojis.dart:23]: I think this should be reverted.

File: tool/update_emojis.dart
Comment (srawlins) [inline tool/update_emojis.dart:34]: I think this should be reverted.

File: tool/update_emojis.dart
Comment (srawlins) [inline tool/update_emojis.dart:51]: I think this should be reverted.

**Review by srawlins — CHANGES_REQUESTED** [review]
The vast majority of the changes improve readability I think.

Just a few where I think the change is much worse.

**Comment by chenzhiguang** [comment]
How about close this PR?  I have just created a new [PR](https://github.com/dart-lang/markdown/pull/419) which has much fewer changes. please have a look.

**Comment by srawlins** [comment]
Closing in favor of #420

**Pull request closed without merging**
